src/main/java/com/eternalcode/discordapp/leveling/LevelController.java

```diff
@@ -73,9 +80,21 @@ public void update(ExperienceChangeEvent experienceChangeEvent) {
                 .register("{level}", String.valueOf(newLevel))
                 .format(this.levelConfig.message.description);
 
-            TextChannel levelChannel = this.jda.getTextChannelById(this.levelConfig.channel);
-            if (levelChannel != null) {
-                levelChannel.sendMessage(messageContent).queue();
+
+            try {
+                Optional<MessageChannel> textChannelOptional = Optional.ofNullable(this.jda.getTextChannelById(event.channelId()));
+
+                MessageChannel channel = textChannelOptional.orElse(this.jda.getPrivateChannelById(event.channelId()));
```

Comment:
Create a separate method that checks whether the channel exists and methods for acquiring channels.
```suggestion
                long channelId = event.getChannelId();

                MessageChannel channel = this.isPrivateChannel(channelId) ? this.getPrivateChannel(channelId) : this.getTextChannel(channelId);
```